src/containers/Body.jsx

```diff
@@ -82,21 +82,27 @@ const Body = ({ data }) => {
           // la key va messa all'elem piu esterno
           <section key={`group-${key}`}>
             <div className={classes.single_date}>{dateToString(elemDate)}</div>
-            {eventData.map((elem) => {
-              return (
-                <Card
-                  key={elem.id}
-                  category={elem.category}
-                  type={elem.type}
-                  where={elem.where}
-                  date={elem.date}
-                  start_time={elem.start_time}
-                  end_time={elem.end_time}
-                  notes={elem.notes}
-                  destination={elem.destination}
-                />
-              );
-            })}
+            {eventData
+              .filter((element) => {
+                return (
+                  element.type === activeFilter && element.type === undefined

```

Comment:
Il problema risiede in questa condizione. Cosa stai chiedendo?

`ritorna true se element.type é uguale ad activeFilter E se element.type é uguale ad undefined`.

Queste due condizioni non possono mai essere vere **contemporaneamente**, perché in questo caso element.type **O** é uguale ad activeFilter **O** é undefined.

Ci sta peró un secondo problema, ovvero la seconda metá della condizione. 

La condizione giusta da utilizzare in questo caso é:
`element.type === activeFilter || !activeFilter`

ovvero:
`ritorna true se element.type é uguale ad activeFilter OPPURE se activeFilter é falsy`

Questo permette di filtrare le card se un filtro é impostato o di vederle tutte se é assente.